util/wait_group_wrapper.go

```diff
@@ -43,7 +43,7 @@ func (w *WaitGroupWrapper) RunWithRecover(exec func(), recoverFn func(r interfac
 	go func() {
 		defer func() {
 			r := recover()
-			if r != nil && recoverFn != nil {
+			if recoverFn != nil {
```

Comment:
`util.WithRecovery` will always call `recoveryFn`, align with it and hashjoin relies on this behavior.